diff --git a/arch/arm/mach-omap2/board-4430sdp.c b/arch/arm/mach-omap2/board-4430sdp.c
index 4fa7ff4..22ea002 100644
--- a/arch/arm/mach-omap2/board-4430sdp.c
+++ b/arch/arm/mach-omap2/board-4430sdp.c
@@ -861,6 +861,7 @@ static int wifi_set_power(struct device *dev, int slot, int power_on, int vdd)
 
 	return 0;
 }
+
 #endif
 
 static struct twl4030_usb_data omap4_usbphy_data = {
@@ -1855,6 +1856,7 @@ static void omap4_4430sdp_wifi_init(void)
 		pr_err("Error setting wl12xx data\n");
 }
 #endif
+
 static void __init omap_4430sdp_init(void)
 {
 	int status;
diff --git a/drivers/mmc/core/core.c b/drivers/mmc/core/core.c
index 777e75e..69d7fc4 100644
--- a/drivers/mmc/core/core.c
+++ b/drivers/mmc/core/core.c
@@ -1370,7 +1370,7 @@ int mmc_suspend_host(struct mmc_host *host)
 	}
 	mmc_bus_put(host);
 
-	if (!err && !(host->pm_flags & MMC_PM_KEEP_POWER))
+	if (!err && !mmc_card_keep_power(host))
 		mmc_power_off(host);
 
 	return err;
@@ -1394,7 +1394,7 @@ int mmc_resume_host(struct mmc_host *host)
 	}
 
 	if (host->bus_ops && !host->bus_dead) {
-		if (!(host->pm_flags & MMC_PM_KEEP_POWER)) {
+		if (!mmc_card_keep_power(host)) {
 			mmc_power_up(host);
 			mmc_select_voltage(host, host->ocr);
 			/*
@@ -1419,6 +1419,10 @@ int mmc_resume_host(struct mmc_host *host)
 			err = 0;
 		}
 	}
+
+	/* clear flag */
+	host->pm_flags &= ~MMC_PM_KEEP_POWER;
+
 	mmc_bus_put(host);
 
 	return err;
diff --git a/drivers/mmc/core/sdio.c b/drivers/mmc/core/sdio.c
index f770d2c..c9e020d 100644
--- a/drivers/mmc/core/sdio.c
+++ b/drivers/mmc/core/sdio.c
@@ -518,7 +518,7 @@ static int mmc_sdio_suspend(struct mmc_host *host)
 		}
 	}
 
-	if (!err && host->pm_flags & MMC_PM_KEEP_POWER) {
+	if (!err && mmc_card_keep_power(host) && mmc_card_wake_sdio_irq(host)) {
 		mmc_claim_host(host);
 		sdio_disable_wide(host->card);
 		mmc_release_host(host);
diff --git a/drivers/mmc/host/omap_hsmmc.c b/drivers/mmc/host/omap_hsmmc.c
index 4569139..e68a3b5 100644
--- a/drivers/mmc/host/omap_hsmmc.c
+++ b/drivers/mmc/host/omap_hsmmc.c
@@ -2511,6 +2511,8 @@ static int __init omap_hsmmc_probe(struct platform_device *pdev)
 	if (mmc_slot(host).nonremovable)
 		mmc->caps |= MMC_CAP_NONREMOVABLE;
 
+	mmc->pm_caps |= MMC_PM_KEEP_POWER;
+
 	omap_hsmmc_conf_bus_power(host);
 
 	res = platform_get_resource_byname(pdev, IORESOURCE_DMA, "tx");
diff --git a/drivers/net/wireless/Makefile b/drivers/net/wireless/Makefile
index 783541a..b0652bc 100644
--- a/drivers/net/wireless/Makefile
+++ b/drivers/net/wireless/Makefile
@@ -50,8 +50,12 @@ obj-$(CONFIG_ATH_COMMON)	+= ath/
 obj-$(CONFIG_MAC80211_HWSIM)	+= mac80211_hwsim.o
 
 obj-$(CONFIG_WL12XX)	+= wl12xx/
+
 obj-$(CONFIG_WL12XX_PLATFORM_DATA)	+= wl12xx/
 
+#obj-$(CONFIG_WL12XX_PLATFORM_DATA)	+= wl12xx/wl12xx_platform_data.o
+
+
 obj-$(CONFIG_IWM)	+= iwmc3200wifi/
 
 obj-$(CONFIG_BCM4329)	+= bcm4329/
diff --git a/include/linux/mmc/host.h b/include/linux/mmc/host.h
index df96fe0..f1a2b7f 100644
--- a/include/linux/mmc/host.h
+++ b/include/linux/mmc/host.h
@@ -328,5 +328,14 @@ static inline int mmc_card_is_powered_resumed(struct mmc_host *host)
 	return host->pm_flags & MMC_PM_KEEP_POWER;
 }
 
+static inline int mmc_card_keep_power(struct mmc_host *host)
+{
+	return host->pm_flags & MMC_PM_KEEP_POWER;
+}
+
+static inline int mmc_card_wake_sdio_irq(struct mmc_host *host)
+{
+	return host->pm_flags & MMC_PM_WAKE_SDIO_IRQ;
+}
 #endif
 
