--- st_kim.c	2011-04-05 10:43:11.922754000 +0300
+++ _st_kim.c	2011-04-03 10:41:29.857369000 +0300
@@ -242,11 +242,11 @@
 
 	if (((struct bts_action *) nxt_action)->type != ACTION_WAIT_EVENT) {
 		pr_err("invalid action after change remote baud command");
-	} else {
+	} else {    /*vic fix*/
 		*ptr = *ptr + sizeof(struct bts_action) +
-			((struct bts_action *)nxt_action)->size;
+			((struct bts_action *)cur_action)->size;
 		*len = *len - (sizeof(struct bts_action) +
-				((struct bts_action *)nxt_action)->size);
+				((struct bts_action *)cur_action)->size);
 		/* warn user on not commenting these in firmware */
 		pr_warn("skipping the wait event of change remote baud");
 	}
@@ -385,6 +385,9 @@
 	}
 	/* fw download complete */
 	release_firmware(kim_gdata->fw_entry);
+    if (len != 0) {
+        pr_err("download_firmware(): ERR, len != 0");
+    }
 	return 0;
 }
 
