--- uim.c	2011-04-04 12:01:46.495109000 +0300
+++ uim.c_hci	2011-03-30 13:35:23.229049000 +0200
@@ -511,14 +511,14 @@
 		UIM_DBG(" Removed gps_drv module");
 	}
 
-	UIM_VER(" Removing fm_drv ");
-	if (rmmod("fm_drv") != 0) {
-		UIM_ERR(" Error removing fm_drv module");
+	UIM_VER(" Removing hci_if_drv ");
+	if (rmmod("hci_if_drv") != 0) {
+		UIM_ERR(" Error removing hci_if_drv module");
 		err = -1;
 	} else {
-		UIM_DBG(" Removed fm_drv module");
+		UIM_DBG(" Removed hci_if_drv module");
 	}
-	UIM_DBG(" Removed fm_drv module");
+	UIM_DBG(" Removed hci_if_drv module");
 
 	UIM_VER(" Removing btwilink ");
 	if (rmmod("btwilink") != 0) {
@@ -547,12 +547,12 @@
 		UIM_DBG(" Removed btwilink module");
 	}
 
-	UIM_VER(" Removing fm_drv ");
-	if (system("rmmod fm_drv") != 0) {
-		UIM_ERR(" Error removing fm_drv module");
+	UIM_VER(" Removing hci_if_drv ");
+	if (system("rmmod hci_if_drv") != 0) {
+		UIM_ERR(" Error removing hci_if_drv module");
 		err = -1;
 	} else {
-		UIM_DBG(" Removed fm_drv module ");
+		UIM_DBG(" Removed hci_if_drv module ");
 	}
 
 	/*Remove the Shared Transport */
@@ -698,15 +698,15 @@
 		UIM_DBG("BT driver built into the kernel ?");
 	}
 
-	if (0 == lstat("/fm_drv.ko", &file_stat)) {
-		if (insmod("/fm_drv.ko", "") < 0) {
-			UIM_ERR(" Error inserting fm_drv module, NO FM? ");
+	if (0 == lstat("/hci_if_drv.ko", &file_stat)) {
+		if (insmod("/hci_if_drv.ko", "") < 0) {
+			UIM_ERR(" Error inserting hci_if_drv module, NO hci_if_drv? ");
 		} else {
-			UIM_DBG(" Inserted fm_drv module");
+			UIM_DBG(" Inserted hci_if_drv module");
 		}
 	} else {
-		UIM_DBG("FM driver module un-available... ");
-		UIM_DBG("FM driver built into the kernel ?");
+		UIM_DBG("hci_if_drv driver module un-available... ");
+		UIM_DBG("hci_if_drv driver built into the kernel ?");
 	}
 
 	if (0 == lstat("/gps_drv.ko", &file_stat)) {
