From 5b911407e3a5276b417c8447f9393c975db42deb Mon Sep 17 00:00:00 2001
From: Eliad Peller <eliad@wizery.com>
Date: Mon, 23 May 2011 13:44:18 +0300
Subject: [PATCH 16/24] set "any" trigger by default

---
 net/wireless/core.c |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/net/wireless/core.c b/net/wireless/core.c
index c22ef34..f6b359f 100644
--- a/net/wireless/core.c
+++ b/net/wireless/core.c
@@ -565,6 +565,16 @@ int wiphy_register(struct wiphy *wiphy)
 			return -EINVAL;
 	}
 
+#ifdef CONFIG_ANDROID
+	/* use wowlan by default */
+	if (rdev->wiphy.wowlan.flags & WIPHY_WOWLAN_ANY) {
+		/* TODO: free wowlan in case we fail later*/
+		rdev->wowlan = kzalloc(sizeof(*rdev->wowlan), GFP_KERNEL);
+		if (!rdev->wowlan)
+			return -ENOMEM;
+		rdev->wowlan->any = true;
+	}
+#endif
 	/* check and set up bitrates */
 	ieee80211_set_bitrate_flags(wiphy);
 
-- 
1.7.1

