diff --git a/include/hardware_legacy/wifi.h b/include/hardware_legacy/wifi.h
index c7b5922..23cc92a 100644
--- a/include/hardware_legacy/wifi.h
+++ b/include/hardware_legacy/wifi.h
@@ -35,22 +35,6 @@ int wifi_load_driver();
  */
 int wifi_unload_driver();
 
-#ifdef OMAP_ENHANCEMENT
-/**
- * Load the Hotspot driver.
- *
- * @return 0 on success, < 0 on failure.
- */
-int hotspot_load_driver();
-
-/**
- * Unload the Hotspot driver.
- *
- * @return 0 on success, < 0 on failure.
- */
-int hotspot_unload_driver();
-#endif /*OMAP_ENHANCEMENT*/
-
 /**
  * Start supplicant.
  *
diff --git a/wifi/wifi.c b/wifi/wifi.c
index c5ed2bc..ec73053 100644
--- a/wifi/wifi.c
+++ b/wifi/wifi.c
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 
+
 #include <stdlib.h>
 #include <fcntl.h>
 #include <errno.h>
@@ -77,18 +78,6 @@ static const char SUPP_CONFIG_TEMPLATE[]= "/system/etc/wifi/wpa_supplicant.conf"
 static const char SUPP_CONFIG_FILE[]    = "/data/misc/wifi/wpa_supplicant.conf";
 static const char MODULE_FILE[]         = "/proc/modules";
 
-#ifdef OMAP_ENHANCEMENT
-static const char AP_DRIVER_MODULE_NAME[]  = "tiap_drv";
-static const char AP_DRIVER_MODULE_TAG[]   = "tiap_drv" " ";
-static const char AP_DRIVER_MODULE_PATH[]  = "/system/etc/wifi/softap/tiap_drv.ko";
-static const char AP_DRIVER_MODULE_ARG[]   = "";
-static const char AP_FIRMWARE_LOADER[]     = "wlan_ap_loader";
-static const char AP_DRIVER_PROP_NAME[]    = "wlan.ap.driver.status";
-static const char HOSTAPD_CONFIG_TEMPLATE[]= "/system/etc/wifi/softap/hostapd.conf";
-static const char HOSTAPD_CONFIG_FILE[]    = "/data/misc/wifi/hostapd.conf";
-#define MAX_BUFFER_SIZE 2048
-#endif
-
 static int insmod(const char *filename, const char *args)
 {
     void *module;
@@ -147,148 +136,6 @@ const char *get_dhcp_error_string() {
     return dhcp_lasterror();
 }
 
-#ifdef OMAP_ENHANCEMENT
-int ensure_hostapd_config_file_exists()
-{
-    char buf[MAX_BUFFER_SIZE];
-    int srcfd, destfd;
-    int nread;
-
-    if (access(HOSTAPD_CONFIG_FILE, R_OK|W_OK) == 0) {
-        return 0;
-    } else if (errno != ENOENT) {
-        LOGE("Cannot access \"%s\": %s", HOSTAPD_CONFIG_FILE, strerror(errno));
-        return -1;
-    }
-
-    srcfd = open(HOSTAPD_CONFIG_TEMPLATE, O_RDONLY);
-    if (srcfd < 0) {
-        LOGE("Cannot open \"%s\": %s", HOSTAPD_CONFIG_TEMPLATE, strerror(errno));
-        return -1;
-    }
-
-    destfd = open(HOSTAPD_CONFIG_FILE, O_CREAT|O_WRONLY, 0660);
-    if (destfd < 0) {
-        close(srcfd);
-        LOGE("Cannot create \"%s\": %s", HOSTAPD_CONFIG_FILE, strerror(errno));
-        return -1;
-    }
-
-    while ((nread = read(srcfd, buf, sizeof(buf))) != 0) {
-        if (nread < 0) {
-            LOGE("Error reading \"%s\": %s", HOSTAPD_CONFIG_TEMPLATE, strerror(errno));
-            close(srcfd);
-            close(destfd);
-            unlink(HOSTAPD_CONFIG_FILE);
-            return -1;
-        }
-        write(destfd, buf, nread);
-    }
-
-    close(destfd);
-    close(srcfd);
-
-    if (chown(HOSTAPD_CONFIG_FILE, AID_SYSTEM, AID_WIFI) < 0) {
-        LOGE("Error changing group ownership of %s to %d: %s",
-             HOSTAPD_CONFIG_FILE, AID_WIFI, strerror(errno));
-        unlink(HOSTAPD_CONFIG_FILE);
-        return -1;
-    }
-    return 0;
-}
-
-static int check_hotspot_driver_loaded() {
-    char driver_status[PROPERTY_VALUE_MAX];
-    FILE *proc;
-    char line[sizeof(AP_DRIVER_MODULE_TAG)+10];
-
-    if (!property_get(AP_DRIVER_PROP_NAME, driver_status, NULL)
-            || strcmp(driver_status, "ok") != 0) {
-        return 0;  /* driver not loaded */
-    }
-    /*
-     * If the property says the driver is loaded, check to
-     * make sure that the property setting isn't just left
-     * over from a previous manual shutdown or a runtime
-     * crash.
-     */
-    if ((proc = fopen(MODULE_FILE, "r")) == NULL) {
-        LOGW("Could not open %s: %s", MODULE_FILE, strerror(errno));
-        property_set(AP_DRIVER_PROP_NAME, "unloaded");
-        return 0;
-    }
-    while ((fgets(line, sizeof(line), proc)) != NULL) {
-        if (strncmp(line, AP_DRIVER_MODULE_TAG, strlen(AP_DRIVER_MODULE_TAG)) == 0) {
-            fclose(proc);
-            return 1;
-        }
-    }
-    fclose(proc);
-    property_set(AP_DRIVER_PROP_NAME, "unloaded");
-    return 0;
-}
-
-int hotspot_load_driver()
-{
-    char driver_status[PROPERTY_VALUE_MAX];
-    int count = 100; /* wait at most 20 seconds for completion */
-
-    if (check_hotspot_driver_loaded()) {
-        return 0;
-    }
-
-    if (ensure_hostapd_config_file_exists() < 0) {
-        LOGE("HotSpot config file error");
-        return -1;
-    }
-
-    if (insmod(AP_DRIVER_MODULE_PATH, AP_DRIVER_MODULE_ARG) < 0)
-        return -1;
-
-    if (strcmp(AP_FIRMWARE_LOADER,"") == 0) {
-        usleep(WIFI_DRIVER_LOADER_DELAY);
-        property_set(AP_DRIVER_PROP_NAME, "ok");
-    }
-    else {
-        property_set("ctl.start", AP_FIRMWARE_LOADER);
-        usleep(WIFI_DRIVER_LOADER_DELAY);
-    }
-    sched_yield();
-    while (count-- > 0) {
-        if (property_get(AP_DRIVER_PROP_NAME, driver_status, NULL)) {
-            if (strcmp(driver_status, "ok") == 0)
-                return 0;
-            else if (strcmp(AP_DRIVER_PROP_NAME, "failed") == 0) {
-                hotspot_unload_driver();
-                return -1;
-            }
-        }
-        usleep(200000);
-    }
-    property_set(AP_DRIVER_PROP_NAME, "timeout");
-    hotspot_unload_driver();
-    return -1;
-}
-
-int hotspot_unload_driver()
-{
-    int count = 20; /* wait at most 10 seconds for completion */
-
-    if (rmmod(AP_DRIVER_MODULE_NAME) == 0) {
-        while (count-- > 0) {
-            if (!check_hotspot_driver_loaded())
-                break;
-            usleep(500000);
-        }
-        if (count) {
-            return 0;
-        }
-        return -1;
-    } else
-        return -1;
-}
-#endif /* OMAP_ENHANCEMENT */
-
 static int check_driver_loaded() {
     char driver_status[PROPERTY_VALUE_MAX];
     FILE *proc;
